server/bleep/src/indexes/doc.rs

```diff
@@ -0,0 +1,997 @@
+        let doc_source = builder.add_text_field("doc_source", STORED);
+        let doc_title = builder.add_text_field("doc_title", TEXT | STORED);
+        let doc_description = builder.add_text_field("doc_description", TEXT | STORED);
+        let relative_url = builder.add_text_field("relative_url", raw.clone());
+        let absolute_url = builder.add_text_field("absolute_url", raw);
+        let header = builder.add_text_field("header", trigram.clone());
+        let ancestry = builder.add_text_field("ancestry", trigram.clone());
+        let text = builder.add_text_field("text", trigram);
+        let start_byte = builder.add_u64_field("start_byte", STORED);
+        let end_byte = builder.add_u64_field("end_byte", STORED);
+        let section_depth = builder.add_u64_field("section_depth", FAST | STORED);
+
+        let raw_relative_url = builder.add_bytes_field("raw_relative_url", FAST | STORED | INDEXED);
+
+        Self {
+            doc_id,
+            point_id,
+            doc_source,
+            doc_title,
+            doc_description,
+            relative_url,
+            absolute_url,
+            header,
+            ancestry,
+            text,
+            start_byte,
+            end_byte,
+            section_depth,
+            raw_relative_url,
+            schema: builder.build(),
+        }
+    }
+}
+
+static STATUS_DONE: &str = "done";
+static STATUS_INDEXING: &str = "indexing";
+
+#[derive(serde::Serialize)]
+pub struct Record {
+    pub id: i64,
+    pub url: String,
+    pub index_status: String,
+    pub name: Option<String>,
+    pub favicon: Option<String>,
+    pub description: Option<String>,
+    pub modified_at: chrono::NaiveDateTime,
+}
+
+#[derive(serde::Serialize, Clone)]
+pub enum Progress {
+    Update(Update),
+    Done(i64),
+}
+
+#[derive(serde::Serialize, Clone)]
+pub struct Update {
+    url: url::Url,
+    discovered_count: usize,
+
+    #[serde(skip)]
+    meta: scraper::Meta,
+}
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("failed to initialize doc index: {0}")]
+    Initialize(String),
+
+    #[error("no document with id: {0}")]
+    InvalidDocId(i64),
+
+    #[error("not a valid docs url: `{0}`")]
+    InvalidUrl(url::Url),
+
+    #[error("failed to parse `{0}` as a url: {1}")]
+    UrlParse(String, url::ParseError),
+
+    #[error("failed to perform sql transaction: {0}")]
+    Sql(#[from] sqlx::Error),
+
+    #[error("io error: {0}")]
+    Io(#[from] std::io::Error),
+
+    #[error("tantivy error: {0}")]
+    Tantivy(#[from] tantivy::error::TantivyError),
+
+    #[error("duplicate url: {0}")]
+    DuplicateUrl(url::Url),
+
+    #[error("network error: {0}")]
+    Network(#[from] reqwest::Error),
+
+    #[error("no docs found at url: {0}")]
+    EmptyDocs(url::Url),
+}
+
+impl Doc {
+    fn index_writer(&self) -> Result<tantivy::IndexWriter, Error> {
+        self.section_index
+            .writer(crate::config::default_buffer_size())
```

Comment:
Rather than using `default_buffer_size` we should use the value calculated here: https://github.com/BloopAI/bloop/blob/81812cb4451833c071fc7ac4529a48166c638549/server/bleep/src/lib.rs#L137C14-L137C14. We've had some issues with the default before.